zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/config/JacksonConfig.java

```diff
@@ -31,4 +34,19 @@ public ObjectMapper objectMapper() {
     gatewayRestObjectMapperCustomizer().accept(builder);
     return builder.build();
   }
+
+  @Bean("yamlObjectMapper")
+  public ObjectMapper yamlObjectMapper() {
+    final var builder = Jackson2ObjectMapperBuilder.yaml();
+    builder.factory(new YAMLFactory());
+    gatewayRestObjectMapperCustomizer().accept(builder);
+    return builder.build();
+  }
+
+  // To avoid conflict between the default ObjectMapper and the custom one (yamlObjectMapper)
+  @Bean
+  public MappingJackson2HttpMessageConverter gatewayRestMessageConverter(
+      final @Qualifier("gatewayRestObjectMapper") ObjectMapper gatewayRestObjectMapper) {
+    return new MappingJackson2HttpMessageConverter(gatewayRestObjectMapper);
+  }
```

Comment:
❓ Could you provide a bit more detail on why this is necessary?

Reply:
I have read https://github.com/camunda/camunda/pull/29628#discussion_r2011750515, but it doesn't clarify it for me yet 😅

Reply:
Yes, MappingJackson2HttpMessageConverter is used by `org.springframework.http.converter.json` (it is a compiled library), and consumes ObjectMapper. 

With the introduction of yamlMapper, the library was having conflict between the 2 declared beans `gatewayRestObjectMapper` and `yamlMapper`. As I cant set `gatewayRestObjectMapper` as primary (dealing problems with tasklist, operate). In another words, the Spring Boot no longer knows which ObjectMapper to use for HTTP (de)serialization, and you have to tell it explicitly.

Reply:
OK, I see 👍 I remember that @danielkelemen ran into issues though in the past with this ([reference](https://github.com/camunda/camunda/issues/23133#issuecomment-2437206465)). If this still creates an issue, we can also leave the YAML Object mapper creation in the transformer, as this should only be created once in the application, if I understand this correctly. And it's a special mapper we don't need anywhere else.

Reply:
Thanks for looking into this 🙏

Reply:
No problems :) I just move the declaration of `yaml` back to the transformer. 
I think its ready to go no

Reply:
Yes, that is probably the easiest fix. 👍
Defining the `MappingJackson2HttpMessageConverter` would cause problems in other scenarios.